### ISSUE TITLE
No enum constant  for RetireJS

### ISSUE THREAD

**Issue by lschnetler** [DependencyTrack/dependency-track#193]
### Issue Type:
  - [x] defect report
  - [ ] enhancement request
  
  
### Current Behavior:
While doing a scan in Jenkins using the Dependency-Check(3.3.1) plugin it finds and publishes 20 vulnerabilities. Trying to push it to the dependency track server is giving issues, it does not show any items or detect any vulnerabilities.

I tried to do an API call to the server, it uploads a 21.9M file to the server. The server then has the following in the logs:
```
09:11:37.869 ERROR [ScanUploadProcessingTask] Error while processing scan result
8/29/2018 11:11:37 AMjava.lang.IllegalArgumentException: No enum constant org.owasp.dependencytrack.model.Vulnerability.Source.RETIREJS
8/29/2018 11:11:37 AM	at java.lang.Enum.valueOf(Enum.java:238)
8/29/2018 11:11:37 AM	at org.owasp.dependencytrack.model.Vulnerability$Source.valueOf(Vulnerability.java:72)
8/29/2018 11:11:37 AM	at org.owasp.dependencytrack.tasks.ScanUploadProcessingTask.processDependency(ScanUploadProcessingTask.java:131)
8/29/2018 11:11:37 AM	at org.owasp.dependencytrack.tasks.ScanUploadProcessingTask.inform(ScanUploadProcessingTask.java:83)
8/29/2018 11:11:37 AM	at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:87)
8/29/2018 11:11:37 AM	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
8/29/2018 11:11:37 AM	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
8/29/2018 11:11:37 AM	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
8/29/2018 11:11:37 AM	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
8/29/2018 11:11:37 AM	at java.lang.Thread.run(Thread.java:748)
```

### Steps to Reproduce (if defect):
1. Scan an Angular 5 project using Dependency-Check
2. Upload to the Dependency-Track server using either Jenkins or API calls
```
echo { > data.json
echo \"projectName\" : \"<project Name>\", >> data.json
echo \"projectVersion\" : \"5.0.0\", >> data.json
echo \"scan\" : \"$( base64 dependency-check-report.xml )\" >> data.json
echo } >> data.json

curl -X PUT \
  <server>/api/v1/scan \
  -H 'Cache-Control: no-cache' \
  -H 'Content-Type: application/json' \
  -H 'X-API-Key: <api key>' \
  -d @data.json
```

### Expected Behavior:
Getting vulnerabilities or count of projects of the project

### Environment:
  
  - Dependency-Track Version: 3.1.0 & 3.1.1
  - Distribution: Docker
  - BoM Format & Version: None
  - Database Server: MySQL
  - Browser: Firefox

### Other Details:

```
8/29/2018 10:58:29 AM08:58:29.985 ERROR [ScanUploadProcessingTask] Error while processing scan result
8/29/2018 10:58:29 AMjava.lang.IllegalArgumentException: No enum constant org.owasp.dependencytrack.model.Vulnerability.Source.RETIREJS
8/29/2018 10:58:29 AM	at java.lang.Enum.valueOf(Enum.java:238)
8/29/2018 10:58:29 AM	at org.owasp.dependencytrack.model.Vulnerability$Source.valueOf(Vulnerability.java:72)
8/29/2018 10:58:29 AM	at org.owasp.dependencytrack.tasks.ScanUploadProcessingTask.processDependency(ScanUploadProcessingTask.java:131)
8/29/2018 10:58:29 AM	at org.owasp.dependencytrack.tasks.ScanUploadProcessingTask.inform(ScanUploadProcessingTask.java:83)
8/29/2018 10:58:29 AM	at alpine.event.framework.BaseEventService.lambda$publish$0(BaseEventService.java:87)
8/29/2018 10:58:29 AM	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
8/29/2018 10:58:29 AM	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
8/29/2018 10:58:29 AM	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
8/29/2018 10:58:29 AM	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
8/29/2018 10:58:29 AM	at java.lang.Thread.run(Thread.java:748)
```

**Comment by stevespringett** [comment]
This was added on August 15th with 734df0ac67b573dc25a8390d4b0833822793c24c and will be included in the upcoming v3.2 release.

**Comment by lschnetler** [comment]
Ok cool thanks for the quick response.
